src/devices/StUsb4500/Objects/PowerDeliveryObject.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using UnitsNet;
+
+namespace Iot.Device.Usb.Objects
+{
+    /// <summary>
+    /// Base class for all the different power delivery objects (=PDO).
+    /// </summary>
+    public abstract class PowerDeliveryObject
+    {
+        /// <summary>Gets or sets the value which encodes all properties of this PDO.</summary>
+        public uint Value { get; protected set; }
+
+        /// <summary>Gets the power of this PDO.</summary>
+        public abstract Power Power { get; }
+
+        /// <summary>Initializes a new instance of the <see cref="PowerDeliveryObject"/> class.</summary>
+        /// <param name="value">The value.</param>
+        protected PowerDeliveryObject(uint value) => Value = value;
+
+        /// <summary>Creates a new PDO from the given value.</summary>
+        /// <param name="value">The value.</param>
+        /// <returns>A PDO of the type defined by the value.</returns>
+        public static PowerDeliveryObject CreateFromValue(uint value)
```

Comment:
1. same comment about name and value documentation
2. you should be consistent: either go with constructor everywhere or Create static method everywhere, if this is an implementation detail it should be protected or internal

Reply:
@krwq @2) The `CreateFromValue` is a helper method if the user does not know the type and then yes, they should check the returned type. If the user knows the type he can directly create the correct type. If this can be made anyway better I'm open for suggestions. Sadly the three possible return types do not share any their properties in all three types...